libs/utils/decorate.js

```diff
@@ -0,0 +1,146 @@
+//
+// Shared decorate utils for Consonant blocks
+// Block usage ref: [media, z-pattern]
+//
+
+import { decorateLinkAnalytics } from './analytics.js';
+
+export function decorateButtons(el, isLarge) {
+  const buttons = el.querySelectorAll('em a, strong a');
+  buttons.forEach((button) => {
+    const parent = button.parentElement;
+    const buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
+    const buttonSize = isLarge ? 'button-XL' : 'button-M';
+    button.classList.add('con-button', buttonType, buttonSize);
+    parent.insertAdjacentElement('afterend', button);
+    parent.remove();
+  });
+  if (buttons.length > 0) {
+    const actionArea = buttons[0].closest('p');
+    actionArea.classList.add('action-area');
+    actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
+  }
```

Comment:
nit: early return instead
```suggestion
  const buttons = el.querySelectorAll('em a, strong a');
  if (buttons.length === 0) return;
  
  buttons.forEach((button) => {
    const parent = button.parentElement;
    const buttonType = parent.nodeName === 'STRONG' ? 'blue' : 'outline';
    const buttonSize = isLarge ? 'button-XL' : 'button-M';
    button.classList.add('con-button', buttonType, buttonSize);
    parent.insertAdjacentElement('afterend', button);
    parent.remove();
  });
  
  const actionArea = buttons[0].closest('p');
  actionArea.classList.add('action-area');
  actionArea.nextElementSibling?.classList.add('supplemental-text', 'body-XL');
```